src/adapters/e4c-ranger-staking.adapter.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright Abridged, Inc. 2022. All Rights Reserved.
+// Node module: @collabland/staking-contracts
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {BindingScope, extensionFor, injectable} from '@loopback/core';
+import {Contract, Provider} from 'ethcall';
+import {BigNumber} from 'ethers';
+import {STAKING_ADAPTERS_EXTENSION_POINT} from '../keys';
+import {BaseStakingContractAdapter, StakingAsset} from '../staking';
+import {E4cRangerStaking__factory} from '../types/factories/E4cRangerStaking__factory';
+
+@injectable(
+  {
+    scope: BindingScope.SINGLETON,
+  },
+  extensionFor(STAKING_ADAPTERS_EXTENSION_POINT),
+)
+export class E4CRangerStakingContractAdapter extends BaseStakingContractAdapter {
+  contractAddress = '0x9c18beA91AE053397918410311dbB89295baE18b';
+
+  supportedAssets: StakingAsset[] = [
+    {
+      asset: 'ERC721:0xbA265B93519E6473F34F46ee35F4B23970F41a3f',
+    },
+  ];
+
+  async getStakedTokenIds(owner: string): Promise<BigNumber[]> {
+    const ecProvider = new Provider();
+    ecProvider.init(this.provider);
+
+    const contract = new Contract(this.contractAddress, E4cRangerStaking__factory.abi);
+
+    const items = new Array(650);
```

Comment:
We don't need 1024 any more? Just curious.

Reply:
> We don't need 1024 any more? Just curious.

650 is enough.